### ISSUE TITLE
Set a rating or keyword for favorited photos

### ISSUE THREAD

**Issue by infused-kim** [RhetTbull/osxphotos#732]
**Is your feature request related to a problem? Please describe.**
One of the most important meta data in my workflow is whether a photo is favorited or not. And it's probably a common need for other people too.

Currently, I don't think there is a built-in option to add this to the exported xmp or exif.

**Describe the solution you'd like**
I propose adding an argument like `--favorite-rating`, which would set a 5-star rating on the photo. This would have the advantage that it would work out of the box with apps that display photo ratings.

Another option is something like `--favorite-keyword` that sets the `is_favorite` keyword.

**Describe alternatives you've considered**
Currently I am using the following option as a workaround:

```toml
keyword_template = [
    "{favorite?is_favorite}",
]
```

**Comment by RhetTbull** [comment]
Thanks for the suggestion.  At one point, I actually had this enabled (the code is still there, just commented out).  I disabled it because it wasn't immediately obvious to me that "favorite" and "star rating" would necessarily mean the same thing to many users. 

https://github.com/RhetTbull/osxphotos/blob/6c1650b7cffefc223374f66012393f14d443fa72/osxphotos/photoexporter.py#L1704-L1706

I've got a long-standing but not yet implemented issue (#243) to add an `--exiftool-template` feature that would allow you to set arbitrary metadata fields using the template system, for example,

`--exiftool-template "Rating" "{favorite?5,}"`

Your suggestion of `--favorite-rating` might be worthwhile and simpler for the average use case but implementing a change like this requires passing around the state in quite a few places so it'll have to be something I do when I have time to get around to it.

**Comment by RhetTbull** [comment]
If I implement this for exif metadata should probably also implement for XMP sidecar.  According to the [XMP spec](https://developer.adobe.com/xmp/docs/XMPNamespaces/xmp/): 

    XMP:Rating    A user-assigned rating for this file. The value shall be -1 or in the range [0..5], where -1 indicates “rejected” 
    and 0 indicates “unrated”. If xmp:Rating is not present, a value of 0 should be assumed. 
    NOTE: Anticipated usage is for a typical “star rating” UI, with the addition of a notion of rejection.

**Comment by infused-kim** [comment]
I agree that it probably shouldn't be enabled by default, but having the ability to enable it through a command line argument would be great.

And yes, support both XMP and exif would be great.

Either way, I don't think it's a super high priority request as my keyword workaround works well enough. But I just wanted to raise the feature request and provide a workaround for those who are looking for this functionality.

**Comment by RhetTbull** [comment]
I've implemented this in [v0.50.9](https://github.com/RhetTbull/osxphotos/releases/tag/v0.50.9) as `--favorite-rating`:

```
--favorite-rating
                                When used with --exiftool or --sidecar, set XMP:Rating=5 for photos marked
                                as Favorite and XMP:Rating=0 for non-Favorites. If not specified, XMP:Rating
                                is not set.
```

**Comment by infused-kim** [comment]
Amazing... it works perfectly.

Thank you so much for implementing it so quickly!
